Source/Particles/Collision/BinaryCollision/PhotonPhoton/PhotonPhotonCollisionFunc.H

```diff
@@ -0,0 +1,243 @@
+/* Copyright 2021 Neil Zaim
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef PHOTON_PHOTON_COLLISION_FUNC_H_
+#define PHOTON_PHOTON_COLLISION_FUNC_H_
+
+#include "SinglePhotonPhotonCollisionEvent.H"
+
+#include "Particles/Collision/BinaryCollision/BinaryCollisionUtils.H"
+#include "Particles/Pusher/GetAndSetPosition.H"
+#include "Particles/MultiParticleContainer.H"
+#include "Particles/WarpXParticleContainer.H"
+#include "Utils/Parser/ParserUtils.H"
+#include "Utils/TextMsg.H"
+#include "WarpX.H"
+
+#include <AMReX_Algorithm.H>
+#include <AMReX_DenseBins.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_Random.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Vector.H>
+
+/**
+ * \brief This functor does two-photon binary collisions on a single cell.
+ *  Particles of the two colliding species are paired with each other and for each pair we compute
+ *  if a pair-production event occurs. If so, we fill a mask (input parameter p_mask) with true so that
+ *  product particles corresponding to a given pair can be effectively created in the particle
+ *  creation functor.
+ *  This functor also reads and contains the event multiplier.
+ */
+class PhotonPhotonCollisionFunc{
+    // Define shortcuts for frequently-used type names
+    using ParticleType = WarpXParticleContainer::ParticleType;
+    using ParticleBins = amrex::DenseBins<ParticleType>;
+    using index_type = ParticleBins::index_type;
+    using SoaData_type = WarpXParticleContainer::ParticleTileType::ParticleTileDataType;
+
+public:
+    /**
+     * \brief Default constructor of the PhotonPhotonCollisionFunc class.
+     */
+    PhotonPhotonCollisionFunc () = default;
+
+    /**
+     * \brief Constructor of the PhotonPhotonCollisionFunc class
+     *
+     * @param[in] collision_name the name of the collision
+     * @param[in] isSameSpecies whether the two colliding species are the same
+     */
+    PhotonPhotonCollisionFunc (const std::string collision_name, MultiParticleContainer const * const /*mypc*/,
+                       const bool isSameSpecies) : m_isSameSpecies(isSameSpecies)
+    {
+        using namespace amrex::literals;
+
+#ifdef AMREX_SINGLE_PRECISION_PARTICLES
+        amrex::Abort("Photon-photon collision module does not currently work with single precision");
+#endif
+
+        amrex::ParmParse pp_collision_name(collision_name);
+        // default event multiplier
+        m_event_multiplier = 1.0_prt;
+        utils::parser::queryWithParser(
+            pp_collision_name, "event_multiplier", m_event_multiplier);
+        // default probability threshold
+        m_probability_threshold = 0.02_prt;
```

Comment:
It might be a bit superfluous but I would suggest writing something like this:
```suggestion
        constexpr auto default_probability_threshold = 0.02_prt;
        m_probability_threshold =default_probability_threshold;
```